internal/confmapprovider/discovery/bundle/bundle.d/receivers/istio.discovery.yaml.tmpl

```diff
@@ -0,0 +1,100 @@
+{{ receiver "prometheus/istio" }}:
+  enabled: true
+  rule:
+    k8s_observer: type == "pod" and ("istio.io/rev" in annotations or labels["istio"] == "pilot" or name matches "istio.*")
```

Comment:
I am not checking for the std `annotations["prometheus.io/scrape"] == "true"` annotation. My thinking is that with discovery mode, we are not expecting users to do the pre-work of adding these, but then on the other hand, this would be an easy way for users to opt-out for certain pods.